src/pages/publication/DescriptionPanel.tsx

```diff
@@ -39,19 +40,17 @@ const StyledFieldHeader = styled.header`
 
 const DescriptionPanel: FC = () => {
   const { t } = useTranslation('publication');
-  const { setFieldTouched, setFieldValue, values }: FormikProps<FormikPublication> = useFormikContext();
+  const { setTouched, touched, setFieldValue, values }: FormikProps<FormikPublication> = useFormikContext();
 
-  // Validation messages won't show on fields that are not touched
-  const setAllFieldsTouched = useCallback(() => {
-    Object.values(DescriptionFieldNames).forEach((fieldName) => setFieldTouched(fieldName));
-  }, [setFieldTouched]);
+  const touchedRef = useRef(touched);
+  useEffect(() => {
+    touchedRef.current = touched;
+  }, [touched]);
 
   useEffect(
-    () => () => {
-      // Set all fields as touched if user navigates away from this panel (on unmount)
-      setAllFieldsTouched();
-    },
-    [setAllFieldsTouched]
+    // Set all fields as touched if user navigates away from this panel (on unmount)
+    () => () => setTouched(mergeTouchedFields([touchedRef.current, touchedDescriptionTabFields])),
+    [setTouched]
   );
```

Comment:
@sarahse Nå har jeg lagt inn dette med egen touchedRef og useEffect for alle tabs.

Burde være mulig å få til noe gjenbruk her. Jeg tenkte noe ala `const _ = useSetTouchedFieldOnUnmount(touchedDescriptionTabFields)`, så kunne den hooken bare hatt en touchedRef og satt aktuelle felt til touched ved unmount, men den vil jo ikke ha noen returverdi da, så det virker så rart... Forslag?

Reply:
Hver tab har altså dette nå:
```
  const touchedRef = useRef(touched);
  useEffect(() => {
    touchedRef.current = touched;
  }, [touched]);

  useEffect(
    () => () => setTouched(mergeTouchedFields([touchedRef.current, <ULIK_PER_TAB>])),
    [setTouched]
  );
```

Reply:
alternativt kan jeg flytte touchedRef ett hakk opp til `PublicationFormTabs`, og sende en setTouched-metode via props, men må fortsatt ha en useEffect per tab da

Reply:
hm. kjipt at dette skulle være så vanskelig da. jeg har ikke noe bedre forslag enn ditt egentlig. jeg må sette meg litt mer inn i det først tror jeg. skal ta ned branchen her og teste litt!

Reply:
kanskje litt mye å sette seg inn i med en gang ja :P tror det vil bli bedre om vi bare flytter touchedRef ett hakk opp

Reply:
det går faktisk ikke, siden komponenten over er utenfor Formik :P 
så må nok lage en ny komponent for dette. Så blir det ca slik i `PublicationForm`:
```
<Form>
   <PublicationFormTabs tabNumber={tabNumber} handleTabChange={handleTabChange} />
   <PublicationFormContent tabNumber={tabNumber} setTouchFields={setTouchFields} />
</Form>
```

Reply:
ah, selvfølgelig. ja, men det er ok 👍